examples/gno.land/r/sys/validators/validators.gno

```diff
@@ -45,33 +49,63 @@ func removeValidator(address std.Address) {
 	}
 
 	// Validator removed, note the change
-	changes = append(changes, change{
+	ch := change{
 		blockNum: std.GetHeight(),
 		validator: validators.Validator{
 			Address:     val.Address,
 			PubKey:      val.PubKey,
 			VotingPower: 0, // nullified the voting power indicates removal
 		},
-	})
+	}
+
+	saveChange(ch)
 
 	// Emit the validator set change
 	std.Emit(validators.ValidatorRemovedEvent)
 }
 
+// saveChange saves the valset change
+func saveChange(ch change) {
+	id := getBlockID(ch.blockNum)
```

Comment:
```suggestion
	id := seqId(getBlockID(ch.blockNum))
```

this way you can iterate with from as key.

Reply:
I'm not sure what you mean by this, the key we use for the `avl.Tree` is a string value, because the API requires it -- the `Iterate` API also requires a string

We actually never have a situation where we want to derive the block number from the actual `seqid.ID`

Reply:
https://github.com/gnolang/gno/blob/master/examples/gno.land/p/demo/seqid/seqid.gno#L64

-> `seqid.ID(42).String()`

no need to use the seqid.Next(), just casting your existing block numbers into seqid.

Reply:
We actually already do this, `getBlockID` does the casting:
https://github.com/gnolang/gno/blob/e62705a91a6b81836899c50edd0d37f887e53e29/examples/gno.land/r/sys/validators/validators.gno#L85-L88

Reply:
Oh, great!

So yeah, now you can use `tree.Iterate(from, to, ...)` where `from, to != "", ""`. (Probably you're already doing).

Reply:
Updated the only traversal as part of an earlier suggestion you made :)

[a027d94](https://github.com/gnolang/gno/pull/2229/commits/a027d940ce48200af7d33f5a61a1987c8aecdead)